From d7574d436d973a44d02e26cc7fa77de21030ad62 Mon Sep 17 00:00:00 2001
From: Christophe Fergeau <cfergeau@redhat.com>
Date: Wed, 2 Apr 2014 13:43:17 +0200
Subject: [PATCH] build-sys: Always prepend '-' to build id

When using the --with-buildid configure paramater, the build id which is
substituted in the MSI wxs file is automatically prepended by a '-', but
the build id which is used in the C files does not get this '-'
automatically.

Currently, the linux and mingw spec files prepend a '-' on their own to the
--with-buildid argument, but this causes the MSI installer to show 2 '-'
during installation: "Please wait while Windows configures VirtViewer
0.6.0--1"

This commit always prepends a '-' to the buildid strings, and removes the
'-' from the spec files. This is to ensure the separator between version
number and buildid is not forgotten, which could give a confusing version
number.
---
 configure.ac              | 9 +++++----
 mingw-virt-viewer.spec.in | 2 +-
 virt-viewer.spec.in       | 2 +-
 3 files changed, 7 insertions(+), 6 deletions(-)

diff --git a/configure.ac b/configure.ac
index 7a00bf4..f966688 100644
--- a/configure.ac
+++ b/configure.ac
@@ -211,10 +211,11 @@ if test "x$have_gtk_vnc" != "xyes" && test "x$have_spice_gtk" != "xyes"; then
 fi
 
 AC_ARG_WITH([buildid],
-    AS_HELP_STRING([--with-buildid=id], [Set additional build version details]))
-AC_DEFINE_UNQUOTED([BUILDID], "$with_buildid", [Build version details])
-if test "x$with_buildid" != x; then
-    AC_SUBST([BUILDID], "-$with_buildid")
+    AS_HELP_STRING([--with-buildid=id], [Set additional build version details]),
+    [buildid="-$with_buildid"], [buildid=""])
+AC_DEFINE_UNQUOTED([BUILDID], "$buildid", [Build version details])
+if test "x$buildid" != x; then
+    AC_SUBST([BUILDID], "$buildid")
 fi
 
 major=`echo $PACKAGE_VERSION | cut -d. -f1`
diff --git a/mingw-virt-viewer.spec.in b/mingw-virt-viewer.spec.in
index 44d9230..fb1c9fd 100644
--- a/mingw-virt-viewer.spec.in
+++ b/mingw-virt-viewer.spec.in
@@ -85,7 +85,7 @@ MinGW Windows virt-viewer MSI
 
 
 %build
-%mingw_configure --with-gtk=2.0 --with-buildid=-%{release}
+%mingw_configure --with-gtk=2.0 --with-buildid=%{release}
 
 %mingw_make %{?_smp_mflags} V=1
 %mingw_make %{?_smp_mflags} V=1 -C data msi
diff --git a/virt-viewer.spec.in b/virt-viewer.spec.in
index 3dfeb96..3cbf194 100644
--- a/virt-viewer.spec.in
+++ b/virt-viewer.spec.in
@@ -109,7 +109,7 @@ autoreconf -if
 %define govirt_arg --with-ovirt
 %endif
 
-%configure %{spice_arg} %{gtk_arg} %{govirt_arg} --with-buildid=-%{release} --disable-update-mimedb
+%configure %{spice_arg} %{gtk_arg} %{govirt_arg} --with-buildid=%{release} --disable-update-mimedb
 %__make %{?_smp_mflags}
 
 
